File: leetcode/cpp/3200.cpp
class Solution
{
public:
    int check(int a, int b)
    {
        int c = 0, f = 0, l = 1;
        while (a > 0 || b > 0)
        {
            if (f == 0)
            {
                if (a >= l)
                {
                    a = a - l;
                }
                else
                    break;
            }
            else
            {
                if (b >= l)
                {
                    b = b - l;
                }
                else
                    break;
            }
            c++;
            l++;
            f = f ^ 1;
        }
        return c;
    }
    int maxHeightOfTriangle(int r, int b)
    {
        int k1 = check(r, b);
        int k2 = check(b, r);
        return max(k1, k2);
    }
};

File: leetcode/python/3200.py
def maxHeightOfTriangle(red: int, blue: int) -> int:
    height1 = 0
    height2 = 0
    balls = 1
    red1 = red
    blue1 = blue
    while(True):
        if blue < balls :
            break
        blue -= balls
        balls+=1
        height1+=1
        if red < balls:
            break
        red -= balls
        balls+=1
        height1+=1
    balls = 1
    while(True):
        if red1 < balls :
            break
        red1 -= balls
        balls+=1
        height2+=1
        if blue1 < balls:
            break
        blue1 -= balls
        balls+=1
        height2+=1
    print(max(height1,height2))
    return max(height1,height2)
    


maxHeightOfTriangle(4,9)
